sandbox/imagery-metadata-extraction_sd2.R

```diff
@@ -32,13 +26,40 @@ process_pitch_values <- function(pitch_values) {
   return(processed_pitch)
 }
 
-# Process each EXIF file
-for (exif_file in exif_files) {
-  exif = prep_exif(exif_file)
+# Define a function to extract datetime and pitch values from EXIF data
+extract_metadata <- function(exif) {
+  # Convert DateTimeOriginal to datetime objects
+  exif$DateTimeOriginal <- ymd_hms(gsub(":", " ", exif$DateTimeOriginal), tz = "UTC")
+
+  # Calculate start and end datetime_local
+  start_datetime_local <- min(exif$DateTimeOriginal)
+  end_datetime_local <- max(exif$DateTimeOriginal)
+
+  # Process pitch values
   camera_pitch_values = exif$CameraPitch
   camera_pitch_derived = process_pitch_values(camera_pitch_values)
```

Comment:
Please modify the function so that it takes `exif` as an argument directly, rather than having to pull the `CameraPitch` attribute first. See my example exif attribute extraction function `extract_flight_speed` [here](https://github.com/open-forest-observatory/ofo-r/blob/4f22ca71bf72078b9307403ad24caf1510f30120/R/imagery-metadata-extraction_dy.R) as an example.